### DISCUSSION TITLE
Difference between the fullhaaboot.bin and haamain.bin ?

### DISCUSSION THREAD

**Discussion by OwenL1** [RavenSystem/esp-homekit-devices#1601] [Q&A] [answered]
Hi everyone, does anyone know what's the difference between the fullhaaboot.bin and haamain.bin in the Binary release? I notice there is a size difference, what about the function difference? And does anyone know how to build the fullhaaboot.bin from haamain.bin?

Thanks

**Comment by RavenSystem** [comment, marked as answer]
fullhaaboot.bin is boot loader + haaboot.bin (HAA OTA Installer stage 1). haamain.bin is main firmware.

**Reply by TienND090** [reply]
And does anyone know how to build the full haa from haamain.bin?
I want to run this setup screen, how do I do that?
Firmware | Setup Color | Description
HAA MAIN | White | Main firmware with HomeKit capabilities

Thanks!

**Reply by RavenSystem** [reply]
You can not run HAAMain without HAABoot. HAAMain needs many tasks performed by HAABoot. Install HAABoot and follow installation instructions.

**Reply by TienND090** [reply]
From the source code I want to generate a full .bin file to use and have a white install screen. I've been searching and can't find any documentation. Hope you help!
![](https://raw.githubusercontent.com/RavenSystem/ravensystem-media/master/setup_white.png)

Thank you very much!

**Reply by RavenSystem** [reply]
You can not do that. You must use fullhaaboot file to install this firmware.

**Reply by TienND090** [reply]
I flashed the fullhaaboot.bin firmware but it shows the orange installation screen

![](https://raw.githubusercontent.com/RavenSystem/ravensystem-media/master/setup_orange.png)

Thanks

**Reply by RavenSystem** [reply]
Yes. It is the installer, the firmware part that manages flash partitions, user settings, flash SPI... needed to run firmware part with white background. Follow installation instructions to finish installation: https://github.com/RavenSystem/esp-homekit-devices/wiki/Installation

**Reply by TienND090** [reply]
I flashed fullhaaboot.bin firmware to esp8266 and used following script but it didn't work
{"c":{"l":13,"b":[{"g":0,"t":5}]},"a":[{"0":{"r":[{ "g":12}]},"1":{"r":[{"g":12,"v":1}]},"b":[{"g":0},{" g":4},{"g":4,"t":0}]}]}
Script in url: https://github.com/RavenSystem/esp-homekit-devices/wiki/Devices-Database

Thanks
